plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -43,6 +46,90 @@ public static boolean isManagedSdkFeatureEnabled() {
     return false;
   }
 
+  private static Object modifyLock = new Object();
+  private static int suspenderCounts = 0;
+  private static boolean isModifyingSdk = false;
+
+  /**
+   * Suspends potential SDK auto-install or auto-update temporarily. This is to use the Cloud SDK
+   * safely by preventing modifying the SDK while using the SDK. If an install or update has already
+   * started, blocks callers until the install or update is complete. Callers must call {@code
+   * CloudSdkManager#allowModifyingSdk} eventually to lift the suspension.
+   *
+   * Any callers that intend to use {@code CloudSdk} must always call this before staring work, even
```

Comment:
Might be less error prone to flip this. E.g. the cloud sdk sets a flag when we start downloading, and it throws an exception or perhaps blocks if you try to do anything while it's downloading.

Reply:
It actually works like that (sort of) at the minimum, without this method. However, I also realized we nees to block the other direction as well: block installing or updating if someone is using the SDK, like deploying. Only the deploy job knows when it is done using the SDK.